cilium/cmd/config.go

```diff
@@ -73,15 +75,42 @@ func configDaemon(cmd *cobra.Command, opts []string) {
 			if err := command.PrintOutput(cfgStatus.Realized.Options); err != nil {
 				os.Exit(1)
 			}
+			if err := command.PrintOutput(cfgStatus.DaemonConfigurationMap); err != nil {
+				os.Exit(1)
+			}
 			return
 		}
+		fmt.Print("### Read-Only Configurations ###\n\n")
+		keys := make([]string, 0, len(cfgStatus.DaemonConfigurationMap))
+		for k := range cfgStatus.DaemonConfigurationMap {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		for _, k := range keys {
+			v := cfgStatus.DaemonConfigurationMap[k]
+			if reflect.ValueOf(v).Kind() == reflect.Map {
+				mapString := make(map[string]string)
+				m, ok := v.(map[string]interface{})
+				if ok {
+					for key, value := range m {
+						mapString[key] = fmt.Sprintf("%v", value)
+					}
+					dumpConfig(mapString)
+					continue
+				} else {
+					fmt.Print("Error: cannot cast daemon config map to map[string]interface{}\n")
+				}
+			}
+			fmt.Printf("%-34s: %v\n", k, v)
+		}
 		dumpConfig(cfgStatus.Immutable)
+		fmt.Print("\n\n##### Read Write Configurations #####\n")
 		dumpConfig(cfgStatus.Realized.Options)
-		fmt.Printf("%-24s %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
-		fmt.Printf("%-24s %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
-		fmt.Printf("%-24s %s\n", "PolicyEnforcement", cfgStatus.Realized.PolicyEnforcement)
+		fmt.Printf("%-34s: %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
+		fmt.Printf("%-34s: %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
```

Comment:
I don't think that these `k8s-configuration`, `k8s-endpoint` options are read/write. In fact in my minikube they don't even print anything:

```
# cilium config 
Conntrack                Enabled
ConntrackAccounting      Enabled
ConntrackLocal           Disabled
Debug                    Disabled
DebugLB                  Disabled
DropNotification         Enabled
MonitorAggregationLevel  Medium
PolicyAuditMode          Disabled
PolicyTracing            Disabled
PolicyVerdictNotification Enabled
TraceNotification        Enabled
k8s-configuration        
k8s-endpoint             
PolicyEnforcement        default
```

We can at least put them in the read-only section, or if someone knows we don't need them, maybe even remove them.

Reply:
Yes, I'll make the necessary changes. For now, I will put them in the read-only section.

Reply:
@joestringer Can we have a small discussion with someone about the unnecessary fields in the cli output?

Reply:
@h3llix I've started a thread here around it: https://cilium.slack.com/archives/C2B917YHE/p1626885955046300

Maybe it's just a bug that we used to print some k8s information for these two fields, but we no longer print it. For now I'm OK with just moving these to read-only, we can work on getting this PR into the tree. Later on if you or others are interested into digging more into why those fields are not printing anything, we can investigate more.